packages/credential-provider-cognito-identity/src/localStorage.spec.ts

```diff
@@ -10,7 +10,12 @@ describe("localStorage", () => {
 
   beforeEach(() => {
     if (window) {
-      (window.localStorage as any) = undefined;
+      Object.defineProperty(window, "localStorage", {
+        writable: true,
+        configurable: true,
+        enumerable: true,
+        value: undefined,
+      });
```

Comment:
this one was likely due to an external bug I mentioned in https://github.com/aws/aws-sdk-js-v3/issues/4807#issuecomment-1583288761. I didn't look too deep into it.